code/projects/app/src/app/stages/e3/e3-diff/e3-diff.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, computed, signal, Signal } from "@angular/core";
+import { toolDifference } from "steg";
+import { ImageComparisonService } from "../../../shared/image-comparison/image-comparison.service";
+import { ImageDisplayComponent } from "../../../shared/image-display/image-display.component";
+import { ImageDownloadComponent } from "../../../shared/image-download/image-download.component";
+import { SliderComponent } from "../slider/slider.component";
+
+@Component({
+  selector: "app-e3-diff",
+  standalone: true,
+  imports: [SliderComponent, ImageDisplayComponent, ImageDownloadComponent],
+  templateUrl: "./e3-diff.component.html",
+  styleUrl: "./e3-diff.component.css",
+})
+export class E3DiffComponent {
+  protected readonly contrast = signal(0);
+  protected readonly brightness = signal(0);
+
+  protected readonly image: Signal<ImageData | undefined>;
+  protected readonly left: Signal<ImageData | undefined>;
+  protected readonly right: Signal<ImageData | undefined>;
+
+  constructor(private readonly imageLocalStorage: ImageComparisonService) {
```

Comment:
I've discussed `inject` above, while it didn't bring almost any benefit there, it would allow you to avoid the entire `constructor` altogether here.